sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -80,14 +137,70 @@ internal class NextLinkOperationImplementation : IOperation
             OperationFinalStateVia finalStateVia,
             string? apiVersion)
         {
-            _requestMethod = requestMethod;
+            AssertNotNull(pipeline, nameof(pipeline));
+            AssertNotNull(requestMethod, nameof(requestMethod));
+            AssertNotNull(startRequestUri, nameof(startRequestUri));
+            AssertNotNull(nextRequestUri, nameof(nextRequestUri));
+            AssertNotNull(headerSource, nameof(headerSource));
+            AssertNotNull(finalStateVia, nameof(finalStateVia));
+
+            RequestMethod = requestMethod;
             _headerSource = headerSource;
             _startRequestUri = startRequestUri;
             _nextRequestUri = nextRequestUri;
             _lastKnownLocation = lastKnownLocation;
             _finalStateVia = finalStateVia;
             _pipeline = pipeline;
             _apiVersion = apiVersion;
+            OperationId = ParseOperationId(nextRequestUri);
+        }
+
+        private static string? ParseOperationId(string nextRequestUri)
+        {
+            if (Uri.TryCreate(nextRequestUri, UriKind.Absolute, out var uri))
+            {
+                return uri.Segments.LastOrDefault();
+            }
+            return null;
+        }
+
+        public RehydrationToken GetRehydrationToken()
+            => GetRehydrationToken(RequestMethod, _startRequestUri, _nextRequestUri, _headerSource.ToString(), _lastKnownLocation, _finalStateVia.ToString(), OperationId);
```

Comment:
Do we know that the fields we're calling `.ToString()` on can never be null?

Reply:
Yes, because we `AssertNotNull` [here](https://github.com/Azure/azure-sdk-for-net/pull/42686/files/adc92770313b373c0e0ed3afd642114d21b1b967#diff-ddf89b49baf148b40544d15990fb244dfa02c26a8a67da24903cb6af0b023e1cR140-R147)